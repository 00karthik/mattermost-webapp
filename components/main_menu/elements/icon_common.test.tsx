// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import IconCommon from './icon_common';

describe('components/main_menu/IconCommon', () => {
    test('should match snapshot', () => {
        const Props = {
            glyph: 'emoticon-happy-outline',
            size: 'xsmall',
            label: 'icon',
        };

        const wrapper = shallow(
            <IconCommon
                {...Props}
            />,
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('label').text()).toBe('icon');
        expect(wrapper.find('glyph').text()).toBe('emoticon-happy-outline');
    });
});
