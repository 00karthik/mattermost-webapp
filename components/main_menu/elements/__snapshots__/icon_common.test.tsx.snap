// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/main_menu/IconCommon should match snapshot 1`] = `
<div
  className="Icon__wrapper"
>
  <i
    aria-label="icon"
    className="Icon Icon___xsmall icon-emoticon-happy-outline"
  />
</div>
`;
